src/backend/core/api/viewsets.py

```diff
@@ -476,15 +475,22 @@ def attachment_upload(self, request, *args, **kwargs):
             return drf_response.Response(
                 serializer.errors, status=status.HTTP_400_BAD_REQUEST
             )
-        # Extract the file extension from the original filename
-        file = serializer.validated_data["file"]
-        extension = os.path.splitext(file.name)[1]
 
         # Generate a generic yet unique filename to store the image in object storage
         file_id = uuid.uuid4()
-        key = f"{document.key_base}/{ATTACHMENTS_FOLDER:s}/{file_id!s}{extension:s}"
+        extension = serializer.validated_data["expected_extension"]
+        key = f"{document.key_base}/{ATTACHMENTS_FOLDER:s}/{file_id!s}.{extension:s}"
+
+        # Prepare metadata for storage
+        metadata = {"Metadata": {"owner": str(request.user.id)}}
+        if serializer.validated_data.get("is_unsafe", False):
```

Comment:
I think this is misleading, because `serializer.validated_data` always has an `is_unsafe` (that may sometimes be `None`).

I would rather use:
```suggestion
        if serializer.validated_data["is_unsafe]:
```